pinot-spi/src/main/java/org/apache/pinot/spi/exception/QueryErrorCode.java

```diff
@@ -153,4 +155,14 @@ public boolean isClientError() {
         return false;
     }
   }
+
+  public boolean isIncludeStackTrace() {
+    switch (this) {
+      case INTERNAL:
+      case UNKNOWN:
+        return true;
+      default:
+        return false;
+    }
+  }
```

Comment:
I'm not sure if including the stack trace should be dependent on the error code, but even if it is, I don't think it is a decision that should be taken here but instead on the class that actually uses this method. Specifically, the one that decides whether the stack should be included is ErrorMseBlock, so IMHO this method should be in that class instead.

Reply:
Now, about when we want to include the stack trace in the error block: We may want to include it if it is an unknown error, but why do we want to do that when it is internal?

Reply:
You're right, moved to a private method in `ErrorMseBlock` itself and including the stack trace only for `UNKNOWN` errors. `INTERNAL` ones should have a message with enough information to know what and where happened.